homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,253 @@
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return (
+            self._details.for_code(frigidaire.HaclCode.TARGET_TEMPERATURE)
+            .containers.for_id(frigidaire.ContainerId.TEMPERATURE)
+            .number_value
+        )
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        convert = {
+            frigidaire.Mode.OFF.value: HVAC_MODE_OFF,
+            frigidaire.Mode.COOL.value: HVAC_MODE_COOL,
+            frigidaire.Mode.FAN.value: HVAC_MODE_FAN_ONLY,
+            frigidaire.Mode.ECO.value: HVAC_MODE_AUTO,
+        }
+
+        frigidaire_mode = self._details.for_code(
+            frigidaire.HaclCode.AC_MODE
+        ).number_value
+
+        return convert[frigidaire_mode]
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return (
+            self._details.for_code(frigidaire.HaclCode.AMBIENT_TEMPERATURE)
+            .containers.for_id(frigidaire.ContainerId.TEMPERATURE)
+            .number_value
+        )
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_COOL,
+            HVAC_MODE_AUTO,
+            HVAC_MODE_FAN_ONLY,
+        ]
+
+    @property
+    def fan_mode(self):
+        """Return the fan setting."""
+        convert = {
+            frigidaire.FanSpeed.OFF.value: FAN_OFF,  # when the AC is off
+            frigidaire.FanSpeed.AUTO.value: FAN_AUTO,
+            frigidaire.FanSpeed.LOW.value: FAN_LOW,
+            frigidaire.FanSpeed.MEDIUM.value: FAN_MEDIUM,
+            frigidaire.FanSpeed.HIGH.value: FAN_HIGH,
+        }
+        fan_speed = self._details.for_code(frigidaire.HaclCode.AC_FAN_SPEED_SETTING)
+
+        if not fan_speed:
+            return FAN_OFF
+
+        return convert[fan_speed.number_value]
+
+    @property
+    def fan_modes(self):
+        """List of available fan modes."""
+        return [
+            FAN_AUTO,
+            FAN_LOW,
+            FAN_MEDIUM,
+            FAN_HIGH,
+        ]
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._appliance.nickname
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 60
+        else:
+            return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 90
+        else:
+            return 32
+
+    @property
+    def unique_id(self):
+        """Return unique ID based on Frigidaire ID."""
+        return self._appliance.appliance_id
```

Comment:
This can be set as entity class attribute, making the code more compact:
<https://developers.home-assistant.io/docs/core/entity#entity-class-attributes>

in the `__init__`:
```py
   self._attr_unique_id = appliance.appliance_id
```